arangod/Scheduler/SupervisedScheduler.h

```diff
@@ -186,16 +196,19 @@ class SupervisedScheduler final : public Scheduler {
   size_t const _maxNumWorkers;
   uint64_t const _maxFifoSizes[NumberOfQueues];
   uint64_t const _ongoingLowPriorityLimit;
+  uint64_t const _maxNumberDetachedThreads;
 
   /// @brief fill grade of the scheduler's queue (in %) from which onwards
   /// the server is considered unavailable (because of overload)
   double const _unavailabilityQueueFillGrade;
 
   std::list<std::shared_ptr<WorkerState>> _workerStates;
   std::list<std::shared_ptr<WorkerState>> _abandonedWorkerStates;
-  std::atomic<uint64_t> _numWorking;  // Number of threads actually working
-  std::atomic<uint64_t> _numAwake;    // Number of threads working or spinning
-                                      // (i.e. not sleeping)
+  std::list<std::shared_ptr<WorkerState>> _detachedWorkerStates;
+  std::atomic<uint64_t> _numWorking;   // Number of threads actually working
+  std::atomic<uint64_t> _numAwake;     // Number of threads working or spinning
+                                       // (i.e. not sleeping)
+  std::atomic<uint64_t> _numDetached;  // Number of detached threads
```

Comment:
I don't see an initializer for `_numDetached` here or in the .cpp file. Doesn't that mean that it has an unspecified value upon initialization?

Reply:
I added a zero initializer in the constructor.